src/utils/wallet.js

```diff
@@ -209,15 +222,39 @@ export class Wallet {
         }
     }
 
-    async createNewAccount(accountId) {
-        this.checkNewAccount(accountId)
+    async createNewAccount(accountId, fundingKey) {
+        this.checkNewAccount(accountId);
+        const keyPair = nearlib.KeyPair.fromRandom('ed25519');
 
-        const keyPair = nearlib.KeyPair.fromRandom('ed25519')
-        await sendJson('POST', CONTRACT_CREATE_ACCOUNT_URL, {
-            newAccountId: accountId,
-            newAccountPublicKey: keyPair.publicKey.toString()
-        })
+        if (fundingKey) {
+            await this.createNewAccountLinkdrop(accountId, fundingKey, keyPair);
+            window.localStorage.removeItem(`nearlib:keystore:${LINKDROP_CONTRACT_ID}:default`)
+            // TODO: Figure out better solution than adding and removing from local storage
+
+        } else {
+            await sendJson('POST', CONTRACT_CREATE_ACCOUNT_URL, {
+                newAccountId: accountId,
+                newAccountPublicKey: keyPair.publicKey.toString()
+            })
+        }
         await this.saveAndSelectAccount(accountId, keyPair);
+
+    }
+
+    async createNewAccountLinkdrop(accountId, fundingKey, keyPair) {
+        const near = await nearlib.connect(fundingConfig);
+        window.localStorage.setItem(`nearlib:keystore:${LINKDROP_CONTRACT_ID}:default`, `ed25519:${fundingKey}`)
+        const contract = await near.loadContract(LINKDROP_CONTRACT_ID, {
+            viewMethods: [],
+            changeMethods: ['create_account_and_claim', 'claim'],
+            sender: LINKDROP_CONTRACT_ID
+        });
+        const publicKey = keyPair.publicKey.toString().split(':')[1]
+        await contract.create_account_and_claim({
+            new_account_id: accountId,
+            new_public_key: publicKey},
+            BOATLOAD_OF_GAS
```

Comment:
this shouldn't be needed, nearlib is AFAIK already passing maximum prepaid gas by default (if it's not – we should update nearlib)